### PR TITLE
feat: STX NFTs Tab

### PR DESCRIPTION
**Pull request by Imamah-Zafar**
# 🔘 PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

# 📜 Background
Provide a brief explanation of why this pull request is needed. Include the problem you are solving or the functionality you are adding. Reference any related issues.

Issue Link: #[[ENG-3140]](https://linear.app/xverseapp/issue/ENG-3140/stx-nft-main-collectible-dashboard-page)
#[[ENG-2807]](https://linear.app/xverseapp/issue/ENG-2807/stacks-nft-tab-data-fetching-pagination-and-load-more-button)

# 🔄 Changes
STX Collectible Dashboard screen UI updated

# 🖼 Screenshot / 📹 Video

https://github.com/secretkeylabs/xverse-web-extension/assets/88320460/f995f411-cc8e-4fb9-b4d0-645281b5ccb4

nft-1

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/components/collectibleCollage/collectibleCollage.tsx [hunk 1]
```diff
@@ -1,4 +1,7 @@
+/* eslint-disable no-nested-ternary */
```
Comment (teebszet) [inline src/app/components/collectibleCollage/collectibleCollage.tsx:1]: should we turn off this eslint rule globally? 😛

File: src/app/utils/inscriptions.ts [hunk 2]
```diff
@@ -53,3 +54,6 @@ export const mapCondensedInscriptionToBundleItem = (
   type: 'inscription',
   rarity_ranking: 'common', // TODO eventually want to fetch this rarity and display it
 });
+
+export const getNftsTabGridItemSubText = (collection: StacksCollectionData) =>
+  collection.total_nft > 1 ? `${collection.total_nft} Items` : '1 Item';
```
Comment (teebszet) [inline src/app/utils/inscriptions.ts:59]: could we move this to utils/nfts.ts ?

File: src/app/screens/sendNft/index.tsx [hunk 3]
```diff
@@ -110,13 +110,10 @@ function SendNft() {
...
   }
   const { nftData } = useNftDataSelector();
-  const nftIdDetails = id!.split('::');
   const [nft, setNft] = useState<NftData | undefined>(undefined);
 
   useEffect(() => {
-    const data = nftData.find(
-      (nftItem) => Number(nftItem?.token_id) === Number(nftIdDetails[2].slice(1)),
-    );
+    const data = nftData.find((nftItem) => nftItem.fully_qualified_token_id === id);
```
Comment (teebszet) [inline src/app/screens/sendNft/index.tsx:116]: repeated code here could also go into utils/nfts.ts

File: src/app/hooks/queries/useStacksCollectibles.ts [hunk 4]
```diff
@@ -1,42 +1,36 @@
...
-  return {
-    isLoading,
-    error,
-    data,
-    fetchNextPage,
-    isFetchingNextPage,
-    hasNextPage,
-    refetch,
-  };
+  return useInfiniteQuery([`nft-collection-data-${stxAddress}`], fetchNfts, {
```
Comment (teebszet) [inline src/app/hooks/queries/useStacksCollectibles.ts:18]:
```suggestion
  return useInfiniteQuery(['nft-collection-data', stxAddress], fetchNfts, {
```

File: src/app/components/collectibleCollage/collectibleCollage.tsx (hunk 1 above)
Comment (Imamah-Zafar) [reply src/app/components/collectibleCollage/collectibleCollage.tsx:1]: would very much like to do so 😅

**Review by teebszet — DISMISSED** [review]

**Review by teebszet — DISMISSED** [review]

**Review by teebszet — APPROVED** [review]

**Pull request merged**